scripts/us_bea/states_gdp/import_data.py

```diff
@@ -0,0 +1,127 @@
+        zip_file = zipfile.ZipFile(io.BytesIO(resp.read()))
+
+        # Open the specific desired file (CSV) from the folder, and decode it.
+        # This results in a string representation of the file. Then clean it
+        # by removing " characters inserted in the decoding process.
+        data = zip_file.open(self.FILE).read().decode('utf-8').replace('"', '')
+
+        # Turn the string representation of the file into a DF.
+        data = [line.split(",") for line in data.split("\n")]
+        df = pd.DataFrame(data[1:], columns=data[0])
+
+        # Filters out columns that are not US States (e.g. New England).
+        # TODO(fpernice): Add non-state entities.
+        df = df[df['GeoName'].isin(US_STATES)]
+
+        # Gets columns that represent quarters, e.g. 2015:Q2.
+        all_quarters = [q for q in df.columns if q[:2] == "20"]
+
+        # Convert table from wide to long format.
+        df = pd.melt(df, id_vars=['GeoFIPS', 'Unit'],
+                     value_vars=all_quarters,
+                     var_name='Quarter')
+
+        qtr_month_map = {
+            'Q1':'03',
+            'Q2':'06',
+            'Q3':'09',
+            'Q4':'12'
+        }
+
+        def date_to_obs_date(date):
+            """Converts date format e.g. 2005:Q3 to e.g. 2005-09."""
+            date = date.replace("\r", "")
+            return date[:4] + "-" + qtr_month_map[date[5:]]
+        df['Quarter'] = df['Quarter'].apply(date_to_obs_date)
+
+        def clean_data_val(data):
+            """Removes '\r' characters in value column and converts to float."""
+            return float(data.replace("\r", ""))
+        df['value'] = df['value'].apply(clean_data_val)
+
+        def convert_geoid(fips_code):
+            """Creates GeoId column. We get lucky that Data Commons's geoIds
+            equal US FIPS state codes.
+            """
+            return "geoId/" + fips_code.replace(" ", "")[:2]
+        df['GeoId'] = df['GeoFIPS'].apply(convert_geoid)
+
+        # Set the instance DF to have one row per geoId/Quarter pair, with
+        # different measurement methods as columns. This facilitates the
+        # design of TMCFs.
+        self.df = df[df['Unit'] == "Millions of chained 2012 dollars"]
+        self.df.set_index(['GeoId', 'Quarter'])
+        self.df["millions_of_chained_2012_dollars"] = self.df['value']
+        quality_indices = df[df['Unit'] == "Quantity index"]
+        self.df["Quantity_index"] = quality_indices['value'].values
+        current_usd = df[df['Unit'] == "Millions of current dollars"]['value']
+        self.df["millions_of_current_dollars"] = current_usd.values
+        self.df = self.df.drop(["GeoFIPS", "Unit", "value"], axis=1)
+
+    def save_csv(self, filename='states_gdp.csv'):
+        """Saves instance DF to specified csv file."""
+        self.df.to_csv(filename)
+
+
+def main():
+    loader = StateGDPDataLoader()
+    loader.save_csv()
+
+
+if __name__ == '__main__':
```

Comment:
can we use the absl app package for the main?

https://abseil.io/docs/python/guides/app